### DISCUSSION TITLE
Newline/Linefeed character?

### DISCUSSION THREAD

**Discussion by SparkyWolf** [ArthurSonzogni/FTXUI#941] [Q&A] [answered]
Ok, bit of a convoluted question here, but the gist of it is this: I'm trying to create a CYOA text game that utilizes FTXUI, and have run into an issue regarding formatting of paragraphs. I need to print multiple, separate paragraphs to a vbox with the click of a button, and have been utilizing a vector of strings (paragraphs) that is indexed through when a button is clicked. This works fine, but I need to print multiple paragraphs simultaneously, separated by white space. I tried using both text() and paragraph() with a newline character, but neither of these functions seem to parse the standard '\n' like cout/printf would. 

Does FTXUI have support for any sort of newline or linefeed character when using the text() or paragraph() functions? If not, is there some relatively simple code that can replicate it?

**Comment by ArthurSonzogni** [comment]
Hello!

What about splitting the text by '\n', then mapping each part into a `paragraph`, and joining them with a `vbox`?

```cpp
#include <ftxui/dom/elements.hpp>  // for paragraph, vbox, Elements
#include <string>                // for string, getline
#include <sstream>               // for stringstream

using namespace ftxui;

Elements splitAndDisplay(const std::string& text) {
  Elements elements;
  std::stringstream ss(text);
  std::string line;

  // Directly create and add paragraph elements while splitting the text
  while (std::getline(ss, line, '\n')) {
    elements.push_back(paragraph(line));
  }

  // Join the paragraphs with a vbox
  return vbox(elements);
}
```

**Reply by SparkyWolf** [reply]
Ok, this is very close to what I need here. Only issue is that I'm looking to print a physical space (equivalent to an empty line of whitespace, or \n\n). Is there any way print something like that? Closest thing I've found in FTXUI is filler(), but I'm not sure that would work here.

**Reply by ArthurSonzogni** [reply, marked as answer]
You could use one of:
```cpp
separatorEmpty()
```
```cpp
separatorStyled(EMPTY)
```
```cpp
emptyElement() | size(HEIGHT, EQUAL, 1)
```

**Reply by SparkyWolf** [reply]
Just had a chance to try this and separatorEmpty() works perfectly, thanks so much for the help!
